### PR TITLE
Add new test for get_pointer_type and get_pointer_device USM API

### PR DESCRIPTION
**Pull request by vasilytric**
Provide verification to cooperation USM functions get_pointer_type
and get_pointer_device with four memory allocation types:
host, device, shared and non USM allocation.

### CODE REVIEW

**Review by ProGTX — APPROVED** [review]

File: tests/usm/usm.h [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    static_assert(alloc != alloc, "Unknown USM allocation type");
+  }
+};
+
+/** @brief Returns an aspect depending on the type of allocated memory
+ *  @tparam alloc USM allocation type
+ *  @retval SYCL aspect that corresponds to allocated memory
+ */
+template <const sycl::usm::alloc alloc>
+auto get_aspect() {
```
Comment (ProGTX) [inline tests/usm/usm.h:52]:
```suggestion
constexpr auto get_aspect() {
```

It also seems like something like this could be useful in the spec ...

**Review by ProGTX — APPROVED** [review]

File: tests/usm/usm.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    static_assert(alloc != alloc, "Unknown USM allocation type");
+  }
+}
+
+/** @brief Return string's description depending on the type of allocated memory
+ *  @tparam alloc USM allocation type
+ *  @retval String description of allocated memory
+ */
+template <sycl::usm::alloc alloc>
+std::string_view get_allocation_description() {
```
[the remark this answers is not in the document]
Comment (vasilytric) [reply tests/usm/usm.h]: I've applied this comment in: https://github.com/KhronosGroup/SYCL-CTS/pull/131/commits/99e4bde4857aa058b01463aa6fdfeec04cb34ba4

File: tests/usm/usm.h [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  } else if constexpr (alloc == sycl::usm::alloc::device) {
+    std::unique_ptr<elems_typeT, decltype(deleter)> usm_memory(
+        sycl::malloc_device<elems_typeT>(num_elements, queue), deleter);
+    return usm_memory;
+  } else if constexpr (alloc == sycl::usm::alloc::host) {
+    std::unique_ptr<elems_typeT, decltype(deleter)> usm_memory(
+        sycl::malloc_host<elems_typeT>(num_elements, context), deleter);
+    return usm_memory;
+  } else {
+    static_assert(alloc != alloc, "Unknown USM allocation type");
```
[the remark this answers is not in the document]
Comment (vasilytric) [reply tests/usm/usm.h:43]: If i correctly understood you.
We using `static_assert` here to interrupt compilation if test developer try to allocate memory from non USM allocation type (here we using all available allocation types from SYCL2020 spec section #4.8.2. "Kinds of unified shared memory")

[the remark this answers is not in the document]
Comment (keryell) [reply tests/usm/usm.h:43]: My question was about
```c++
static_assert(false, "Unknown USM allocation type");
```
that you cannot probably use?

[the remark this answers is not in the document]
Comment (vasilytric) [reply tests/usm/usm.h]: Yes, that good decision, thanks for your advice
This change was made in: https://github.com/KhronosGroup/SYCL-CTS/pull/131/commits/684a2ec24693c5aa64fd6b853c69c981ef9e3c9e

**Review by bader — APPROVED** [review]

File: tests/usm/usm.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuriykoch) [reply tests/usm/usm.h]: @keryell @vasilytric I assume 684a2ec will break compilation due non-conditional static_assert. See [SO thread](https://stackoverflow.com/questions/38304847/constexpr-if-and-static-assert) for details

[the remark this answers is not in the document]
Comment (vasilytric) [reply tests/usm/usm.h:43]: @keryell Yes, `static_assert(false, "Unknown USM allocation type");` cannot be used

**Review by keryell — APPROVED** [review]

**Pull request merged**